### ISSUE TITLE
Navbar: Search Overlay input can't be cleared with `Esc` Key

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2149]
The Navbar Search Overlay input can't be cleared with <kbd>Esc</kbd> key.

Demo: https://modus-web-components.trimble.com/?path=/story/components-navbar--default&args=showSearch:true;enableSearchOverlay:true

This should be simple to fix by adding type="search" to the modus-text-input within the search-box.
